### PR TITLE
Update ktlint.version from 0.39.0 to 0.43.0

### PR DESCRIPTION
**Pull request by Cheshiriks**
### What's done:
* Updates ktlint.version from 0.39.0 to 0.43.0
* Increased minimum required gradle version to 5.3
* Fix how file path is passed to Params in diktat-maven-plugin
* gradle plugin with absolute paths doesn' work on Windows and Mac (windows - because of short directory names like `C:\Users\MYUSER~1\...` that can't be properly resolved in Java) - so **forbid** absolute path in *gradle plugin*
* Disable standard ruleset in CLI test
* Add tests for CLI execution with absolute paths

### CODE REVIEW

**Comment by petertrr** [comment]
Looking at the tests, it seems that since gradle metadata has seen its first stable release [in 5.3](https://blog.gradle.org/gradle-metadata-1.0), we should increase a minimum required gradle version and increase the version in tests. 5.0, which we have now, is pretty outdated now anyway.
Btw, actual error is because class Bundling [is available since 5.3](https://docs.gradle.org/current/javadoc/org/gradle/api/attributes/Bundling.html).

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/KotlinParser.kt [hunk 1]
```diff
@@ -27,6 +27,7 @@ import sun.reflect.ReflectionFactory
 /**
  * A class that wraps kotlin compiler's code parser and converts source code into AST
  */
+@Suppress("EMPTY_BLOCK_STRUCTURE_ERROR")
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/KotlinParser.kt:30]: Why? It should be fine with current snapshot

Comment (Cheshiriks) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/KotlinParser.kt]: I probably haven't updated my project.

Comment (Cheshiriks) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/KotlinParser.kt]: No, the project failed on this error

**Comment by Cheshiriks** [comment]
Now userData["file_path"] in Ktlint contains only file name. This makes full path not available for rules.

issue https://github.com/pinterest/ktlint/issues/1133

**Review by akuleshov7 — CHANGES_REQUESTED** [review]
It looks like we need to think about the move to the newest version of ktlint or some other framework, because diktat cli is working incorectly with ktlint 0.39 (not able to disable standard ruleset)

**Comment by petertrr** [comment]
@Cheshiriks ktlint 0.42.0 is out. Please update this PR, bump version and see, if any of our problems are solved in the new version (IIRC, file patterns should work better now)

**Comment by petertrr** [comment]
We should also disable standard rulesetes in our test for CLI execution

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**